apx/common/src/apx_nodeData.c

```diff
@@ -372,6 +389,11 @@ int8_t apx_nodeData_writeInPortData(apx_nodeData_t *self, const uint8_t *src, ui
    else
    {
       memcpy(&self->inPortDataBuf[offset], src, len);
+      if (self->inPortDirtyFlags != 0)
+      {
+         // Flag that in data has been written (And might differ from default value)
+         memset(&self->inPortDirtyFlags[offset], 1, len);
```

Comment:
You seem to have repurposed the inPortDirtyFlags into using them as flags used only during init data creation.
This was never the original purpose of the flags.
Be aware that the original clearing of the flags are still implemented in the function apx_nodeData_readInPortData. This double-use of the flags are highly confusing.
Shouldn't the clearing of the flags in apx_nodeData_readInPortData be removed to clear out confusuin?

Reply:
I could not see that there was a previous use for the flags so i found one :) The port will not be read before the entire file is sent to the node - right, so it should not matter if the flags are cleared. For the new pupose of the flags there is no need to ever clear the flags (except perhaps if a node disconnects?)

Maybe the flags should be renamed to better reflect the new use of them. inPortSetByProviderFlags perhaps?